File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RegularExpressions/RegexShouldNotContainAdjecentWhitespace.vb
Imports System.ComponentModel.DataAnnotations
Imports System.Text.RegularExpressions

Class Compliant
    Private Sub Ctor()
        Dim defaultOrder = New Regex("some pattern", RegexOptions.None)
        Dim namedArgs = New Regex(options:=RegexOptions.None, pattern:="some pattern")
    End Sub

    Private Sub [Static]()
        Dim isMatch = Regex.IsMatch("some input", "some pattern", RegexOptions.None)
    End Sub

    <RegularExpression("[0-9]+")>
    Public Property Attribute As String
End Class

Class Noncompliant
    Private Sub Ctor()
        Dim patternOnly = New Regex("A  B") ' Noncompliant {{The pattern contains adjacent whitespace.}}
        '                           ^^^^^^
    End Sub

    Private Sub [Static]()
        Dim isMatch = Regex.IsMatch("some input", "A  B") ' Noncompliant
        Dim match = Regex.Match("some input", "A  B") ' Noncompliant
        Dim matches = Regex.Matches("some input", "A  B") ' Noncompliant
        Dim replace = Regex.Replace("some input", "A  B", "some replacement") ' Noncompliant
        Dim split = Regex.Split("some input", "A  B") ' Noncompliant
    End Sub

    <RegularExpression("A  B")> ' Noncompliant
    Public Property Attribute As String
End Class

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RegularExpressions/RegexShouldNotContainAdjecentWhitespace.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;

class Compliant
{
    void Ctor()
    {
        var defaultOrder = new Regex("some pattern", RegexOptions.None); // Compliant

        var namedArgs = new Regex(
            options: RegexOptions.None,
            pattern: "some pattern");
    }

    void Static()
    {
        var isMatch = Regex.IsMatch("some input", "some pattern", RegexOptions.None); // Compliant
    }

    [RegularExpression("[0-9]+")] // Compliant
    public string Attribute { get; set; }
}

class Noncompliant
{
    void Ctor()
    {
        var patternOnly = new Regex("A  B"); // Noncompliant {{The pattern contains adjacent whitespace.}}
        //                          ^^^^^^
    }

    void Static()
    {
        var isMatch = Regex.IsMatch("some input", "A  B"); // Noncompliant
        //                                        ^^^^^^
        var match = Regex.Match("some input", "A  B"); // Noncompliant
        var matches = Regex.Matches("some input", "A  B"); // Noncompliant
        var replace = Regex.Replace("some input", "A  B", "some replacement"); // Noncompliant
        var split = Regex.Split("some input", "A  B"); // Noncompliant
    }

    [RegularExpression("A  B")] // Noncompliant
    public string Attribute { get; set; }
}
